sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -54,7 +55,16 @@ class AzureAppConfigurationClient:
     # pylint:disable=protected-access
 
     def __init__(self, base_url, credential, **kwargs):
-        # type: (str, AppConfigConnectionStringCredential, dict) -> None
+        # type: (str, any, dict) -> None
+        try:
+            if not base_url.lower().startswith('http'):
+                base_url = "https://" + base_url
```

Comment:
I don't know where the inputs to this are coming from, is there a chance protocol-relative URLs e.g. `//some.url` could be passed in?

Reply:
The valid urls are like
https://fake_app_config.azconfig-test.io
http://fake_app_config.azconfig-test.io   (though we will raise error in credential policy)
fake_app_config.azconfig-test.io   (we want to add https:// for them)